cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/Job.java

```diff
@@ -101,8 +102,9 @@ void checkTransition(State nextState) {
   boolean isToBeDeleted();
 
   /**
-   * @return the time at which this job was marked for deletion
+   * @return the time at which this job was marked for deletion, null only if {@link #isToBeDeleted} returns false
    */
-  long getToBeDeletedTimestamp();
+  @Nullable
+  Long getToBeDeletedTimestamp();
```

Comment:
I see. How about `getDeleteTimeMillis`? I like to have `millis` to be consistent with `System.currentTimeMillis()` that returns timestamp in milliseconds.